docs/development/justfile.md

```diff
@@ -0,0 +1,769 @@
+
+# Option 2: Activate venv first
+source ../.venv/bin/activate
+source ../.env.local
+pytest ../enterprise/backend/tests/
+```
+
+### Loading and Clearing Environment Variables
+
+Use these just recipes to manage environment variables in your shell:
+
+```bash
+# Load .env.local into current shell
+eval "$(just env-load)"
+
+# Clear all variables from .env.local
+eval "$(just env-clear)"
+
+# Alternative: start a fresh shell (loses all changes)
+exec bash
+```
+
+**Note:** The recipes output commands that you `eval` because just runs in a subshell and can't modify your parent shell directly.
+
+### When to Use Each Approach
+
+| Use `just` when... | Use direct commands when... |
+|-------------------|----------------------------|
+| Running standard tasks | You need tab completion for paths |
+| Sharing commands with teammates | Debugging/exploring interactively |
+| CI/CD scripts | You know the exact command already |
+| You don't need completion | IDE integration |
+
+### Installing just Completions
+
+While `just` can't complete *arguments* to the underlying commands, it can complete recipe names:
+
+```bash
+# Bash
+just --completions bash > ~/.local/share/bash-completion/completions/just
+
+# Zsh
+just --completions zsh > ~/.zfunc/_just
+# Add to ~/.zshrc: fpath=(~/.zfunc $fpath)
+
+# Fish
+just --completions fish > ~/.config/fish/completions/just.fish
+```
+
+After installation, `just b <TAB>` will complete recipe names like `test`, `lint`, `run-dev-server`, etc.
+
+## Personal Recipes
+
+You can add your own custom recipes that aren't tracked in git. This is useful for personal shortcuts, project-specific commands, or experimental recipes.
+
+### Setup
+
+```bash
+# Copy the example file
+cp local.just.example local.just
+
+# Edit with your own recipes
+vim local.just
+```
+
+The `local.just` file is automatically imported by the main justfile (using `import? 'local.just'`). Your recipes will appear in `just --list` alongside the standard recipes.
+
+### Example local.just
+
+```just
+# Personal recipes - not tracked in git
+
+# Quick test for a specific module I'm working on
+my-test:
+    just b test tests/baserow/contrib/builder/
+
+# My preferred dev startup (services only, no frontend)
+my-dev:
+    just dc-dev up -d redis db mailhog
+    just b run-dev-server
+
+# Reset my local database
+my-reset:
+    just dc-dev exec db psql -U baserow -c "DROP SCHEMA public CASCADE; CREATE SCHEMA public;"
+    just b migrate
+```
+
+### Per-Component Recipes
+
+You can also create `local.just` in component directories:
+
+- `backend/local.just` - Backend-specific personal recipes
+- `web-frontend/local.just` - Frontend-specific personal recipes
+
+Each component's justfile imports its own `local.just` independently.
+
+### Notes
+
+- All `local.just` files are gitignored
+- Use `local.just.example` in the project root as a starting template
+- Your recipes can call other recipes using `just <recipe>`
+
+## Troubleshooting
```

Comment:
A bit unsure about the general troubleshooting section here. It describes some possible errors but doesn't even mention if they are for docker or local workflow. I'd personally just scratch it here. If we want to keep some common problems, better to keep them in the documents for each workflow.